### ISSUE TITLE
group_by suddenly broken

### ISSUE THREAD

**Issue by tonk** [ansible/ansible#2277]
After a `git pull` this morning my `group_by` tasks are suddenly broken.

In my playbook I have

```
- hosts: all
  tasks:
      - name: Create an OS group by family
        group_by: key=${ansible_os_family}

- hosts: Debian
  tasks:
      - name: Install certificate tools
        apt: pkg=ca-certificates state=latest

      - name: Install apt-tools
        apt: pkg=python-software-properties state=latest
```

This worked yesterday.

Since the last update I get:

```
TASK: [Create an OS group by family] ********************* 
fatal: [jabber] => Traceback (most recent call last):
  File "/usr/lib/python2.6/site-packages/ansible/runner/__init__.py", line 297, in _executor
    exec_rc = self._executor_internal(host)
  File "/usr/lib/python2.6/site-packages/ansible/runner/__init__.py", line 354, in _executor_internal
    return self._executor_internal_inner(host, self.module_name, self.module_args, inject, port)
  File "/usr/lib/python2.6/site-packages/ansible/runner/__init__.py", line 482, in _executor_internal_inner
    result = handler.run(conn, tmp, module_name, module_args, inject, self.complex_args)
  File "/usr/lib/python2.6/site-packages/ansible/runner/action_plugins/group_by.py", line 43, in run
    args.update(utils.parse_kv(module_args))
NameError: global name 'utils' is not defined


FATAL: all hosts have already failed -- aborting
```

Seems something broke somewhere.

**Comment by dhozac** [comment]
Thanks for the report!

**Comment by tonk** [comment]
Please reopen. It's not broken anymore, but it also stopped working.

In my playbook:

```
- hosts: all
  tasks:
      - name: Create an OS group by family
        group_by: key=${ansible_os_family}

- hosts: RedHat
  tasks:
      - debug: msg="No license available for RedHat servers"

- hosts: Debian
  tasks:
      - debug: msg="In the Debian branch. Family => $ansible_os_family"

      - name: Install Python apt tools
        apt: pkg=python-apt state=latest
```

And running `ansible-playbook` with `-vvv --forks 1` I get:

```
TASK: [Create an OS group by family] ********************* 
<jabber> ESTABLISH CONNECTION FOR USER: root on PORT 22 TO jabber
created 'group_by' ActionModule: key=Debian
changed: [jabber] => {"changed": true, "groups": {"Debian": ["jabber", "master", "puppet"]}, "item": ""}

PLAY [RedHat] ********************* 
skipping: no hosts matched

PLAY [Debian] ********************* 

TASK: [debug msg="In the Debian branch. Family => $ansible_os_family"] ********************* 
<jabber> ESTABLISH CONNECTION FOR USER: root on PORT 22 TO jabber
ok: [jabber] => {"msg": "In the Debian branch. Family => Debian"}
<master> ESTABLISH CONNECTION FOR USER: root on PORT 22 TO master
ok: [master] => {"msg": "In the Debian branch. Family => RedHat"}
<puppet> ESTABLISH CONNECTION FOR USER: root on PORT 22 TO puppet
ok: [puppet] => {"msg": "In the Debian branch. Family => Debian"}
```

The RedHat server says it's the RedHat OS family but Ansible tries to run the playbook part for Debian.
I believe that to be wrong.

**Comment by mpdehaan** [comment]
I'll revert the invalidation patch.  one sec.

**Comment by mpdehaan** [comment]
Reverted in dc0515a, let me know if that works for you.

**Comment by mpdehaan** [comment]
Actually Daniel fixed in 5d82b4b1b776054af654fdab21a56f86c0552ea8
